test/integration/flags.go

```diff
@@ -31,19 +32,55 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+var startTimeout = flag.Int("timeout", 25, "number of minutes to wait for minikube start")
 var binaryPath = flag.String("binary", "../../out/minikube", "path to minikube binary")
 var globalArgs = flag.String("minikube-args", "", "Arguments to pass to minikube")
 var startArgs = flag.String("minikube-start-args", "", "Arguments to pass to minikube start")
 var mountArgs = flag.String("minikube-mount-args", "", "Arguments to pass to minikube mount")
 var testdataDir = flag.String("testdata-dir", "testdata", "the directory relative to test/integration where the testdata lives")
+var disableParallel = flag.Bool("disable-parallel", false, "run the tests squentially and disable all parallel runs")
 
 // NewMinikubeRunner creates a new MinikubeRunner
-func NewMinikubeRunner(t *testing.T, extraArgs ...string) util.MinikubeRunner {
+func NewMinikubeRunner(t *testing.T, profile string, extraStartArgs ...string) util.MinikubeRunner {
 	return util.MinikubeRunner{
-		BinaryPath: *binaryPath,
-		StartArgs:  *startArgs + " " + strings.Join(extraArgs, " "),
-		GlobalArgs: *globalArgs,
-		MountArgs:  *mountArgs,
-		T:          t,
+		Profile:      profile,
+		BinaryPath:   *binaryPath,
+		StartArgs:    *startArgs + " " + strings.Join(extraStartArgs, " "),
+		GlobalArgs:   *globalArgs,
+		MountArgs:    *mountArgs,
+		TimeOutStart: time.Duration(*startTimeout) * time.Minute,
+		T:            t,
 	}
 }
+
+// isTestNoneDriver checks if the current test is for none driver
+func isTestNoneDriver() bool {
+	return strings.Contains(*startArgs, "--vm-driver=none")
+}
+
+// profileName chooses a profile name based on the test name
+// to be used in minikube and kubecontext across that test
+func profileName(t *testing.T) string {
+	if isTestNoneDriver() {
```

Comment:
This function seems simple enough, but all test helpers (functions which accept *testing.T) should call immediately `t.Helper()`, so that Go can attribute failures in the test helper to the line where the helper was called.